packages/trader/src/App/Containers/populate-header.jsx

```diff
@@ -18,15 +18,23 @@ const PopulateHeader = observer(() => {
         onClickCancel: onPositionsCancel,
     } = portfolio;
 
-    const filtered_positions = positions.filter(
-        p =>
-            p.contract_info &&
-            symbol === p.contract_info.underlying &&
-            (trade_contract_type.includes('turbos')
-                ? filterByContractType(p.contract_info, TURBOS.SHORT) ||
-                  filterByContractType(p.contract_info, TURBOS.LONG)
-                : filterByContractType(p.contract_info, trade_contract_type))
-    );
+    const filtered_positions = positions.filter(p => {
+        if (p.contract_info && symbol === p.contract_info.underlying) {
+            if (trade_contract_type.includes('turbos')) {
+                return (
+                    filterByContractType(p.contract_info, TURBOS.SHORT) ||
+                    filterByContractType(p.contract_info, TURBOS.LONG)
+                );
+            }
+            if (trade_contract_type.includes('vanilla')) {
+                return (
+                    filterByContractType(p.contract_info, VANILLALONG.CALL) ||
+                    filterByContractType(p.contract_info, VANILLALONG.PUT)
+                );
+            }
+            return filterByContractType(p.contract_info, trade_contract_type);
+        }
+    });
```

Comment:
```suggestion
    const filtered_positions = positions.filter(
        p =>
            p.contract_info &&
            symbol === p.contract_info.underlying &&
            (isTurbosContract(trade_contract_type) || isVanillaContract(trade_contract_type)
                ? filterByContractType(p.contract_info, isTurbosContract(trade_contract_type) ? TURBOS.SHORT : VANILLALONG.CALL) ||
                  filterByContractType(p.contract_info, isTurbosContract(trade_contract_type) ? TURBOS.LONG : VANILLALONG.PUT)
                : filterByContractType(p.contract_info, trade_contract_type))
    );
```
please check whether or not the shorter version above will work :)

Reply:
Good, point! It's working, thank you